src/storage/disk_manager.c

```diff
@@ -6786,6 +6804,23 @@ disk_sectors_to_extend_npages (const int num_pages)
   return DISK_SECTS_ROUND_UP (DISK_PAGES_TO_SECTS (num_pages));
 }
 
+DKNVOLS
+disk_get_perm_volume_count ()
+{
+  assert (disk_Cache != NULL);
+  return disk_Cache->nvols_perm;
+}
+
+void
+disk_set_page_server_perm_volume_count (DKNVOLS nvols)
+{
+  assert (is_tran_server_with_remote_storage ());
+
+  // Set the number of permanent volumes for transaction server with remote storage.
+  // Disk manager is not initialized yet, so save the number to be used later when disk cache is loaded.
+  disk_Page_server_perm_volume_count = nvols;
```

Comment:
assert that `disk_Page_server_perm_volume_count` is 0 at this point?

Reply:
done